src/reader/DifficultyFeedbackBox.js

```diff
@@ -12,25 +11,30 @@ import SentimentNeutralOutlinedIcon from '@mui/icons-material/SentimentNeutralOu
 
 let FEEDBACK_OPTIONS = { "Easy": 1, "Ok": 3, "Difficult": 5 };
 
-export default function DifficultyFeedbackBox({ api, articleID }) {
-  const [answerSubmitted, setAnswerSubmitted] = useState(false);
+export default function DifficultyFeedbackBox({ api, articleInfo, setArticleInfo, setAnswerSubmitted, source }) {
   const [isHovered, setIsHovered] = useState(false);
 
-  function submitAnswer(answer) {
+  function updateArticleDifficultyFeedback(answer) {
+    let newArticleInfo = { ...articleInfo, relative_difficulty: answer};
     api.submitArticleDifficultyFeedback(
-      { article_id: articleID, difficulty: answer },
+      { article_id: articleInfo.id, difficulty: answer },
       () => {
         setAnswerSubmitted(true);
+        setArticleInfo(newArticleInfo);
       }
     );
+    api.logReaderActivity(api.DIFFICULTY_FEEDBACK, articleInfo.id, answer, source);
   }
 
-  if (answerSubmitted) {
-    return (
-      <s.InvisibleBox>
-        <h3 align="center">Thank You {random(["🤗", "🙏", "😊", "🎉"])}</h3>
-      </s.InvisibleBox>
-    );
+  const diffToOption = (diff) => {
```

Comment:
rename this please to `difficulty`. diff is well-defined in CS, and the first thought i have is about diff-ing files. 

also, it seems to be less a file than a constant.  

also, it should never be undefined. so that should not need to be covered. or?

Reply:
The *diff(iculty)ToOption* function is used with *articleInfo.relative_difficulty*, which could be undefined if the user has not given difficulty-feedback for the given article. Have we misunderstood what relative_difficulty shoud be?

But it is true that if not of the if-statements return true, then the function would return undefined automatically. We will remove it 😊